.pre-commit-config.yaml

```
@@ -1,10 +1,26 @@
 repos:
 - repo: https://github.com/astral-sh/ruff-pre-commit
   # Ruff version.
-  rev: v0.4.10
+  rev: v0.5.0
   hooks:
     # Run the linter.
     - id: ruff
       args: [ --fix ]
     # Run the formatter.
     - id: ruff-format
+
+- repo: local
```

If the intention of this pr is expanding pre-commit hooks, I'd like to suggests additional hooks as used by the nvda dd-on template:

```suggestion
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.6.0
    hooks:
      - id: check-ast
      - id: check-case-conflict
      - id: check-merge-conflict
      - id: end-of-file-fixer
- repo: local
```

---

I think maybe we should also consider `trailing-whitespace`.

Thanks for catching these


```suggestion
- repo: https://github.com/pre-commit/pre-commit-hooks
  rev: v4.6.0
  hooks:
    - id: check-ast
    - id: check-case-conflict
    - id: check-merge-conflict
    - id: end-of-file-fixer
    - id: trim-trailing-whitespace

- repo: local
```